sdk/storage/src/table/continuation_token.rs

```diff
@@ -0,0 +1,163 @@
+use azure_core::errors::AzureError;
+use http::HeaderMap;
+use std::borrow::Cow;
+use url::Url;
+
+const HEADER_NEXTPARTITIONKEY: &str = "x-ms-continuation-NextPartitionKey";
+const HEADER_NEXTROWKEY: &str = "x-ms-continuation-NextRowKey";
+const QUERY_PARAM_NEXTPARTITIONKEY: &str = "NextPartitionKey";
+const QUERY_PARAM_NEXTROWKEY: &str = "NextRowKey";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ContinuationToken {
+    pub(crate) previous_url: Url,
+    pub(crate) new_url: Url,
+}
+
+impl ContinuationToken {
+    pub fn new(previous_url: Url, next_partition_key: &str, next_row_key: Option<&str>) -> Self {
+        let mut partition_key_replaced = false;
+        let mut row_key_replaced = false;
+
+        let v: Vec<(_, _)> = previous_url
+            .query_pairs()
+            // filter_map allows us to strip the QUERY_PARAM_NEXTROWKEY
+            // if next_row_key is empty.
+            .filter_map(|(k, v)| {
+                match k.as_ref() {
+                    QUERY_PARAM_NEXTPARTITIONKEY => {
+                        partition_key_replaced = true;
+                        Some(Cow::Borrowed(next_partition_key))
+                    }
+                    QUERY_PARAM_NEXTROWKEY => {
+                        row_key_replaced = true;
+                        next_row_key.map(|next_row_key| Cow::Borrowed(next_row_key))
+                    }
+                    _ => Some(v),
+                }
+                .map(|new_v| (k, new_v))
+            })
+            .collect();
+
+        let mut new_url = previous_url.clone();
+        new_url.query_pairs_mut().clear().extend_pairs(v);
+        if !partition_key_replaced {
+            new_url
+                .query_pairs_mut()
+                .append_pair(QUERY_PARAM_NEXTPARTITIONKEY, &next_partition_key);
+        }
+        if !row_key_replaced {
+            if let Some(next_row_key) = next_row_key {
+                new_url
+                    .query_pairs_mut()
+                    .append_pair(QUERY_PARAM_NEXTROWKEY, &next_row_key);
+            }
+        }
+
+        Self {
+            previous_url,
+            new_url,
+        }
+    }
+
+    pub fn previous_url(&self) -> &Url {
```

Comment:
The fact that these are public methods makes me think that the TODO for getting the correct host for the URL should probably be resolved, otherwise this could be very confusing to a user when they suddenly get a dummy URL out.

Reply:
Yes the dummy was a bad idea! It should fixed in the last push though!

Reply:
![image](https://user-images.githubusercontent.com/431595/99951918-1c127400-2d7f-11eb-87f4-d613c9c538b4.png)